File: bayesian.py
# http://www.cs.cmu.edu/~tom/mlbook/NBayesLogReg.pdf
from pandas import read_csv
import numpy as np

train = read_csv('train.csv')

test = train.iloc[31500:]
train = train.iloc[:31500]

digits_p = {}
for i in range(10):
    d = train[train['label']==i].values[:,1:]
    digits_p[i] = np.mean(d, 0)

n = len(test)
test['guess'] = np.tile(np.nan, n)
for i in range(n):
    print i
    item = test.iloc[i].values[1:-1]
    m = {}
    for k in range(10):
        m[k] = 1
        for j in range(len(item)):
            m[k] = m[k] * digits_p[k][j]

File: bayesian.jl
using DataFrames

data = readtable("train.csv");
test = data[1:nrow(data)/10, :]
train = data[nrow(data)/10:, :]

digits_p = Dict()
for i = 1:10
    i = i - 1
    d = train[train["label"] .== i, 2:]
    digits_p[i] = Dict()
    for j = 1:ncol(d)
        for k 
    end
end

guess = []
n = nrow(test)
